Library/Homebrew/extend/os/mac/extend/ENV/super.rb

```diff
@@ -113,7 +113,7 @@ def set_x11_env_if_installed
   def setup_build_environment(formula = nil)
     generic_setup_build_environment(formula)
     self["HOMEBREW_SDKROOT"] = effective_sysroot
-    self["SDKROOT"] = MacOS.sdk_path if MacOS::Xcode.without_clt?
+    self["SDKROOT"] = MacOS.sdk_path if MacOS.sdk_path
```

Comment:
Setting SDKROOT is undesirable on older systems that have /usr/include available as we don't want to bring back the 10.11 clock_gettime and 10.12 futimens, etc. problems from the dead.

Reply:
Maybe `self["SDKROOT"] = MacOS.sdk_path if MacOS.sdk_path && !Mac::CLT.provides_headers?`?

Reply:
Yes, that should definitely be safer. I'm not sure sure if that's the behavior we'll end up wanting on Mojave, though.

Reply:
I think it would be great if we could avoid setting it on the backend at all, including in the no-usr-include and xcode-only cases, and leave it to formulae to set when it fixes something. When we were always setting it, conversations with upstream would often get confused because build success/failure that occurred when building within Homebrew would sometimes not occur outside of Homebrew unless SDKROOT was set.

Reply:
Updated to experimentally remove this line entirely. I'll do some testing to see how reliable it is.